stumpy/stump.py

```diff
@@ -190,6 +190,25 @@ def _compute_diagonal(
                 if T_B_subseq_isconstant[i + k] and T_A_subseq_isconstant[i]:
                     pearson = 1.0
 
+                # due to imprecision, the value of pearson may be outside
+                # of interval [-1, 1]
+                if pearson >= 1.0:
+                    pearson = 1.0
+                if pearson <= -1.0:
+                    pearson = -1.0
+
+                if pearson != 1.0 and pearson >= config.STUMPY_CORRELATION_THRESHOLD:
+                    # refine pearson only when we have to
+                    if pearson > ρ[thread_idx, i, 0] or (
+                        ignore_trivial and np.any(pearson > ρ[thread_idx, i + k])
+                    ):
+
+                        d = np.linalg.norm(
```

Comment:
Technically, instead of computing `d` with `norm`, we really need `D_squared`. So, I think we should avoid `norm` and compute the squared distance directly if possible and avoid the costly `sqrt`

Reply:
Basically, you'd need:

```
# x = np.empty(m)
# y = np.empty(m)
x[:] = T_A[i : i + m] - μ_Q[i]) * σ_Q_inverse[i]
y[:] =  (T_B[i + k : i + k + m] - M_T[i + k]) * Σ_T_inverse[i + k]
D_squared = np.sum(np.square(x)) + np.sum(np.square(y)) - 2.0 * np.dot(x, y)
pearson = 1.0 - 0.5 * m_inverse * D_squared
```

Reply:
Nice! I will go with your suggestion.

Reply:
I hope it is obvious that `x` and `y` would only be created once and reused many times. I think `T_A_subseq` and `T_B_subseq` are better names than `x` and `y` though but I think you get the point

Reply:
Btw, the precision of `D_squared` can be wrong and so you'll want to check:

```
if D_squared < config.STUMPY_P_NORM_THRESHOLD:
    D_squared = 0.0
pearson = 1.0 - 0.5 * m_inverse * D_squared
```

Reply:
I created x and y outside of for-loop and re-used them. I will change their names to what you suggested.

Also, thanks for the heads up about the precision of D_squared. I will keep that in mind.